packages/grafana-prometheus/src/querybuilder/components/MetricCombobox.tsx

```diff
@@ -86,9 +90,14 @@ export function MetricCombobox({
   }, [onGetMetrics]);
 
   const metricsExplorerEnabled = config.featureToggles.prometheusMetricEncyclopedia;
-  const inlineFieldWidth = 20;
-  const buttonSize: ComponentSize = 'md';
-  const styles = useStyles2(getMectricComboboxStyles, inlineFieldWidth, buttonSize);
+
+  const theme = useTheme2();
+  const { height } = getPropertiesForButtonSize(BUTTON_SIZE, theme);
+  const buttonSpace = parseInt(theme.spacing(height), 10);
+  const inlineFieldSpace = parseInt(theme.spacing(INLINE_FIELD_WIDTH), 10);
+  const widthToSubstract = inlineFieldSpace + buttonSpace;
```

Comment:
You can put all this in `getMetricComboboxStyles`, then you don't need to use `useTheme2` here